### ISSUE TITLE
Adding a class only to the snapped slide

### ISSUE THREAD

**Issue by davidjerleke** [davidjerleke/embla-carousel#542]
### Discussed in https://github.com/davidjerleke/embla-carousel/discussions/374

<div type='discussions-op-text'>

<sup>Originally posted by **jln13x** September 15, 2022</sup>
Hey there,

first of all: Embla is amazing, thanks for your work!

I constantly ran into the requirement to only style the currently snapped slide (usually the one in the center) and the plugin `embla-carousel-class-names` couldn't help with that. The closest option would be the `selected` one however that applies to all visible slides which is not something that I want.

Use Case: 
![image](https://user-images.githubusercontent.com/85513960/190501723-95b4fa9b-756f-481c-839d-80c2f6b142a9.png)

So I went ahead and quickly wrote a [plugin](https://github.com/jln13x/embla-carousel-snap-class) for it and published it to npm to reuse it across multiple projects. The plugin basically just adds a class and allows me to style it.

Would that be something for the `embla-carousel-class-names`-Plugin ?
If so, I would go ahead and contribute to it and delete my stuff. 

If not, is it ok for me to keep the plugin published? Anything I should add?

Thanks alot! 
</div>

### Specification

```ts
const options = {
  inView: 'is-in-view', // Applied to slides in view
  snapped: 'is-snapped' // Applied to all slides that belong to the selected scroll snap
}
```

@jln13x

**Comment by davidjerleke** [comment]
To be released with **v8.0.0-rc12**.

**Comment by davidjerleke** [comment]
@jln13x this feature has been released with [v8.0.0-rc12](https://github.com/davidjerleke/embla-carousel/releases/tag/v8.0.0-rc12).

**Comment by HarshArora-1205** [comment]
Hi @davidjerleke.

Can you elaborate on how to use classnames (if possible) with tailwind css?

**Comment by davidjerleke** [comment]
@HarshArora-1205 I don’t understand what you want to do? Please share some details about what you want to achieve.

**Comment by HarshArora-1205** [comment]
Okay, so, in runtime the plugin adds class `is-snapped` and `is-in-view` to the slides.

When I am using Tailwind css, I wanted to tweak certain properties like opacity and other props by tracking `is-snapped` class. This can be done by writing custom classes or corresponding js. But I thought if there is any solution to work with snapped classes directly using tailwind css.

So, I stumbled upon this thread in tailwind discussions:
https://github.com/tailwindlabs/tailwindcss/discussions/7282

Thus I got to a workaround:
Adding class `group` on embla__slides and on their child adding class `opacity-0 group-[.is-snapped]:opacity-100`

This worked well and my query got resolved. 
My query was more related to tailwind classes and prefixes but I wonder if there can be a Tailwind class based support in Embla carousel?

**Comment by davidjerleke** [comment]
@HarshArora-1205 if I'm understanding you correctly, you can configure the snapped class with the [snapped option](https://www.embla-carousel.com/plugins/class-names/#snapped):

```tsx
ClassNames({ snapped: 'my-tailwind-class-goes-here' })
```

Is this what you're looking for?

**Comment by HarshArora-1205** [comment]
Uhm no. default class is okay for me, but when an active slide gets the class is snapped then I want that slide to have full opacity.

This can be done by using custom classes and rules. But like we have hover:opactity-100 class in tailwind to set opacity on hover, is there any way to access if is snapped class is set on the slide or not.

Something like [.is-snapped]:opacity-100.
There are workarounds for this which I found. Like creating tailwind plugin or using group.


Anyways, as I cleared earlier this was more related to tailwind than embla.
